bin/crew

```diff
@@ -718,13 +667,12 @@ def upgrade(*pkgs, build_from_source: false)
         return false
       end
     end
-    pkgVer_latest     = Package.load_package(pkgFile, pkgName).version
+    set_package File.join(CREW_PACKAGES_PATH, "#{pkgName}.rb")
     pkgVer_installed  = @device[:installed_packages].select { |pkg| pkg[:name] == pkgName } [0][:version]
-    pkgHash_latest    = Package.load_package(pkgFile, pkgName).get_binary_sha256(@device[:architecture])
     pkgHash_installed = @device[:installed_packages].select { |pkg| pkg[:name] == pkgName } [0][:binary_sha256]
 
-    return pkgHash_latest != pkgHash_installed unless !pkgHash_installed || pkgHash_latest == ''
-    return pkgVer_latest != pkgVer_installed
+    return @pkg.binary_sha256&.dig(ARCH.to_sym) != pkgHash_installed unless !pkgHash_installed || @pkg.binary_sha256&.dig(ARCH.to_sym) == 'null'
```

Comment:
Does `@pkg.binary_sha256&[ARCH.to_sym]` works? (`.dig` seems unnecessary in this case)

Reply:
I'm not sure-- this particular piece of code is very fragile, so I'm trying to avoid changing behaviour

Reply:
It's passing the github action, which does an install, so it works well enough to do that...